py/emitnative.c

```diff
@@ -1591,6 +1606,17 @@ static void emit_native_load_subscr(emit_t *emit) {
                             asm_rv32_opcode_lw(emit->as, REG_RET, reg_base, index_value << 2);
                             break;
                         }
+                        #elif N_XTENSA || N_XTENSAWIN
+                        if (index_value > 0) {
+                            if (index_value < 16) {
+                                asm_xtensa_op_l32i_n(emit->as, REG_RET, reg_base, index_value);
+                                break;
+                            }
+                            if (index_value < 256) {
+                                asm_xtensa_op_l32i(emit->as, REG_RET, reg_base, index_value);
+                                break;
+                            }
```

Comment:
I suggest:
```c
if (index_value > 0 && index_value < 256) {
    asm_xtensa_l32i_optimised(emit->as, REG_RET, reg_base, index_value);
}
```
just to make it a little simpler.

Also, similar comment to the s32i below.

Reply:
Missed that, thanks!  That should be sorted out now.